packages/router-api/src/msg.rs

```diff
@@ -25,23 +26,25 @@ pub enum ExecuteMsg {
      * Router Admin Methods
      * All the below messages should only be called by the router admin
      */
-    // Freezes a chain, in the specified direction.
-    FreezeChain {
-        chain: ChainName,
-        direction: GatewayDirection,
+    /// If no filter is provided, freezes all chain connections in every direction. Should only be called in emergencies.
+    /// If the filter is set, only the specified chains will be frozen in the specified directions.
+    FreezeChains {
+        chains: Option<HashMap<ChainName, GatewayDirection>>,
     },
```

Comment:
I actually would prefer to have a specific `FreezeAllChains` msg, instead of wrapping the argument in an Option. Then `FreezeChains` and `UnfreezeChains` can be symmetrical, and have the same parameters. I am not sure that we actually gain anything by combining freeze all here, since it complicates the parameter.